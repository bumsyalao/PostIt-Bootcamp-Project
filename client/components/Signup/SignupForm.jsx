import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router';
import { withRouter } from 'react-router-dom';

class SignupForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password:'',
      phonenumber:'',
      confirmPassword:'',
      invalid: false
    }
    this.state = { loggedIn: false};
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  onSubmit(event) {
    event.preventDefault();
    this.props.userSignupRequest(this.state)
      .then(() => {
        this.setState({ loggedIn: true });
        this.props.history.push('/homepage');
      })
  }
    render() {
      {this.state.loggedIn && <Redirect to="/" />}
      return (
        <div className="form-margin">
        <form className="col s12 container">
          <div className="input-field col s6 offset-s3">
              <i className="material-icons prefix">account_circle</i>
              <input value={this.state.username} onChange={this.onChange} name="username"  type="text" 
              className="validate" required/>
              <label className="active" htmlFor="username">username</label>
            </div>
            <div className="input-field col s6 offset-s3">
              <i className="material-icons prefix">email</i>
              <input value={this.state.email} onChange={this.onChange} name="email" type="email" 
              className="validate" required/>
              <label className="active" htmlFor="email">email</label>
            </div>
            <div className="input-field col s6 offset-s3">
              <i className="material-icons prefix">phone</i>
              <input value={this.state.phonenumber} onChange={this.onChange} name="phonenumber"type="tel" class="validate" required/>
              <label className="active" htmlFor="icon_telephone">phone number</label>
            </div>
            <div className="input-field col s6 offset-s3">
              <i className="material-icons prefix">lock</i>
              <input value={this.state.password} onChange={this.onChange} name="password"type="password" 
              className="validate" required/>
              <label className="active" htmlFor="password">password</label>
            </div>
            <div className="input-field col s6 offset-s3">
              <i className="material-icons prefix">lock</i>
              <input value={this.state.confirmPassword} onChange={this.onChange} name="confirmPassword" type="password" 
              className="validate" required/>
              <label className="active" htmlFor="confirmPassword">confirm password</label>
            </div>
          <button onClick={this.onSubmit} disabled={this.state.invalid} className="btn waves-effect waves-light col s6 offset-s3 red lighten-2" type="submit" name="action">Register
            <i className="material-icons right">send</i>
          </button>
        </form>
        </div>
      );
    }
  }


SignupForm.propTypes = {
  userSignupRequest: React.PropTypes.func.isRequired,
}

export default withRouter(SignupForm);