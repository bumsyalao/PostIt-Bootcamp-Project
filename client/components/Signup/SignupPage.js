import React from 'react';

class SignupPage extends React.component {
    render() {
        return (
    <section class="container">
        <div class="active">
        <div class="row">
            <form class="col s12">
            <div class="row">
                <div class="input-field col s6">
                <i class="material-icons prefix">account_circle</i>
                <input placeholder="Username" id="user_name" type="text" class="validate">
                <!--<label for="user_name">User Name</label>-->
            </div>
            </div>
            <div class="row">
                <div class="input-field col s6">
                <i class="material-icons prefix">email</i>
                <input placeholder="Email" id="email" type="email" class="validate">
                <!--<label for="email">Email</label>-->
                </div>
            </div>
            <div class="row">
                <div class="input-field col s6">
                <i class="material-icons prefix">vpn_key</i>
                <input placeholder="Password" id="password" type="password" class="validate">
                <!--<label for="password">Password</label>-->
                </div>
            </div>
            <div class="row">
                <div class="input-field col s6">
                <i class="material-icons prefix">vpn_key</i>
                <input placeholder="Confirm password" id="password" type="password" class="validate">
                <!--<label for="password">Confirm Password</label>-->
                </div>
            </div>
            <button class="btn waves-effect waves-light col s6 red lighten-2" type="submit" name="action">Register
                <i class="material-icons right">send</i>
            </button>
            </form>
        </div>
        </div>
        </section>
        );
    }
}

export default SignupPage;