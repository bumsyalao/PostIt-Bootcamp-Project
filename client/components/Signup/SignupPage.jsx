import React from 'react';
import SignupForm from './SignupForm';
import { connect } from 'react-redux';
import { userSignupRequest } from '../../actions/signupAction';
import NavigationBar from '../NavigationBar';


class SignupPage extends React.Component {
  constructor(props){
    super(props)
  }
    render() {
        const { userSignupRequest, addFlashMessage } = this.props;
        return (
          <div>
            <NavigationBar />
            <div className="row">
               <div className="col-md-4 col-md-offset-4">
                   <SignupForm userSignupRequest={userSignupRequest} />
                </div>
            </div> 
            </div>
        );
    }
}
const mapDispatchToProps = (dispatch) => {
  return {
    userSignupRequest:(user) => dispatch(userSignupRequest(user))
  }
};

SignupPage.propTypes = {
  userSignupRequest: React.PropTypes.func.isRequired,
}

export default connect(null, mapDispatchToProps)(SignupPage);
