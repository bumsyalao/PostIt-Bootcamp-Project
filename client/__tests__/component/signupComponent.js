
/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import adapter from 'enzyme-adapter-react-16';
import { SignupForm } from '../../components/Signup/SignupForm';

configure({ adapter: new adapter() });

describe('SignUpForm Component', () => {
  const spy = jest.fn();
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  afterEach(() => {
    spy.mockReset();
  });
  const props = {
    match: {
      path: ''
    },
    access: {
      isAuthenticated: false
    },
    history: {
      push: spy
    },
    userSignupRequest: jest.fn(() => Promise.resolve())
  };
  const component = shallow(
    <SignupForm {...props} />
  );
  test('should match the SignUpForm snapshot', () => {
    const tree = toJson(component);
    expect(component.find('input').length).toEqual(5);
    expect(tree).toMatchSnapshot();
  });
  test('it should set state when onChange function is called', () => {
    component.instance().onChange(
      { target: { value: 'banku', id: 'username' } });
    expect(component.state('username')).toEqual('banku');
  });
  test(
    'it should submit fields in state when onSubmit function is called', () => {
      jest.spyOn(component.instance(), 'onSubmit');
      component.instance().onChange(
      { target: { value: 'banku', id: 'username' } });
      component.instance().onChange(
      { target: { value: 'banku@email.com', id: 'email' } });
      component.instance().onChange(
      { target: { value: '09082091930', id: 'phonenumber' } });
      component.instance().onChange(
      { target: { value: 'banku123', id: 'password' } });
      component.instance().onChange(
      { target: { value: 'banku123', id: 'confirmPassword' } });
      component.find('#submit-signup').simulate('click', { preventDefault: () => {} });
      expect(component.find('#submit-signup').length).toEqual(1);
      expect(component.instance().onSubmit.mock.calls.length).toEqual(0);
    });
  test('onSubmit function should run', () => {
    component.instance().onChange(
      { target: { value: 'banku', id: 'username' } });
    component.instance().onChange(
      { target: { value: 'banku@email.com', id: 'email' } });
    component.instance().onChange(
      { target: { value: '09082091930', id: 'phonenumber' } });
    component.instance().onChange(
      { target: { value: 'banku123', id: 'password' } });
    component.instance().onChange(
      { target: { value: 'banku123', id: 'confirmPassword' } });
    expect(component.state('username')).toEqual('banku');
    expect(component.state('email')).toEqual('banku@email.com');
    expect(component.state('phonenumber')).toEqual('09082091930');
    expect(component.state('password')).toEqual('banku123');
    expect(component.state('confirmPassword')).toEqual('banku123');
    const newspy = jest.spyOn(component.instance(), 'onSubmit');
    component.find('#submit-signup').simulate('click', { preventDefault: () => {} });
    expect(newspy).toHaveBeenCalled();
  });
});
