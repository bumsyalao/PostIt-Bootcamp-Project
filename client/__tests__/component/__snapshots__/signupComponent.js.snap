// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm Component should match the SignUpForm snapshot 1`] = `
<div
  className="form-margin"
>
  <form
    className="col s12 container"
  >
    <div
      className="input-field col s6 offset-s3"
    >
      <i
        className="material-icons prefix"
      >
        account_circle
      </i>
      <input
        className="validate"
        id="username"
        name="username"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        className="active"
        htmlFor="username"
      >
        username
      </label>
    </div>
    <div
      className="input-field col s6 offset-s3"
    >
      <i
        className="material-icons prefix"
      >
        email
      </i>
      <input
        className="validate"
        id="email"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
      <label
        className="active"
        htmlFor="email"
      >
        email
      </label>
    </div>
    <div
      className="input-field col s6 offset-s3"
    >
      <i
        className="material-icons prefix"
      >
        phone
      </i>
      <input
        className="validate"
        id="phoneNumber"
        name="phoneNumber"
        onChange={[Function]}
        required={true}
        type="tel"
        value=""
      />
      <label
        className="active"
        htmlFor="icon_telephone"
      >
        phone number
      </label>
    </div>
    <div
      className="input-field col s6 offset-s3"
    >
      <i
        className="material-icons prefix"
      >
        lock
      </i>
      <input
        className="validate"
        id="password"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <label
        className="active"
        htmlFor="password"
      >
        password
      </label>
    </div>
    <div
      className="input-field col s6 offset-s3"
    >
      <i
        className="material-icons prefix"
      >
        lock
      </i>
      <input
        className="validate"
        id="confirmPassword"
        name="confirmPassword"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
      <label
        className="active"
        htmlFor="confirmPassword"
      >
        confirm password
      </label>
    </div>
    <button
      className="btn waves-effect waves-light col s6 offset-s3 red lighten-2"
      id="submit-signup"
      name="action"
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
